repo.go

```diff
@@ -1062,3 +1080,17 @@ func (r *Repo) timestampFileMeta(roleFilename string) (data.TimestampFileMeta, e
 	}
 	return util.GenerateTimestampFileMeta(bytes.NewReader(b), r.hashAlgorithms...)
 }
+
+func (r *Repo) Payload(roleFilename string) ([]byte, error) {
+	role := strings.TrimSuffix(roleFilename, ".json")
+	if !roles.IsTopLevelRole(role) {
+		return nil, ErrInvalidRole{role, "only signing top-level metadata supported"}
+	}
+
+	s, err := r.SignedMeta(roleFilename)
```

Comment:
should this make sure this is json canonicalized?
it seems like it's just json unmarshalled into data.Signed
https://github.com/theupdateframework/go-tuf/blob/0e25bb50a25b479318f82432dc61ec5461c28322/repo.go#L675-L678

It seems like in most places we set the meta to be the canonicalized form (https://github.com/theupdateframework/go-tuf/blob/0e25bb50a25b479318f82432dc61ec5461c28322/repo.go#L511) but what if someone uses a local store and edits the metadata retrieved?

Maybe I should open a separate issue about this, because I'm not convinced we canonicalize consistently.

Reply:
See my comments in #247. I do think you're broadly right.

That said, I think there's enough in-flight at the moment that we should probably open a separate issue and figure out a consistent strategy for canonicalization there. I'll go ahead and do that.

Reply:
https://github.com/theupdateframework/go-tuf/issues/261